server/game/cards/14.1-RaW/CityOfTheRichFrog.js

```diff
@@ -0,0 +1,37 @@
+const ProvinceCard = require('../../provincecard.js');
+const AbilityDsl = require('../../abilitydsl');
+const { Phases, Players } = require('../../Constants');
+
+class CityOfTheRichFrog extends ProvinceCard {
+    setupCardAbilities() {
+        this.persistentEffect({
+            condition: () => this.game.currentPhase !== 'setup',
+            effect: AbilityDsl.effects.refillProvinceTo(3)
+        });
+
+        this.persistentEffect({
+            targetController: Players.Self,
+            effect: AbilityDsl.effects.playerDelayedEffect({
+                when: {
+                    onPhaseEnded: event => event.phase === Phases.Setup
+                },
+                message: '{0} fills to 3 cards!',
+                messageArgs: effectContext => [effectContext.source],
+                gameAction: AbilityDsl.actions.sequential([
+                    AbilityDsl.actions.moveCard(context => ({
+                        target: context.player.dynastyDeck.first(),
+                        destination: context.source.location
+                    })),
+                    AbilityDsl.actions.moveCard(context => ({
+                        target: context.player.dynastyDeck.first(),
+                        destination: context.source.location
+                    }))
+                ])
+            })
+        });
```

Comment:
Would much rather have a simple action for this than having to manual this for everything that is going to fill a province to a certain x cards.